odl/solvers/util/callback.py

```diff
@@ -53,62 +50,62 @@ def __call__(self, result):
     def __and__(self, other):
         """Return ``self & other``
 
-        Compose partials, calls both in sequence.
+        Compose callbacks, calls both in sequence.
 
         Parameters
         ----------
-        other : `Partial`
+        other : `Partial` or `callable`
             The other partial to compose with
 
         Returns
         -------
-        result : `Partial`
+        result : `SolverCallback`
             A partial whose `__call__` method calls both constituents
             partials.
 
         Examples
         --------
-        >>> store = StorePartial()
-        >>> iter = PrintIterationPartial()
+        >>> store = CallbackStore()
+        >>> iter = CallbackPrintIteration()
         >>> both = store & iter
         >>> both
-        StorePartial() & PrintIterationPartial()
+        CallbackStore() & CallbackPrintIteration()
         """
-        return AndPartial(self, other)
+        return _CallbackAnd(self, other)
 
     def __repr__(self):
         """Return ``repr(self)``"""
         return '{}()'.format(self.__class__.__name__)
 
 
-class AndPartial(Partial):
+class _CallbackAnd(SolverCallback):
 
-    """Partial used for combining several partials"""
+    """Callback used for combining several callbacks"""
 
     def __init__(self, *callbacks):
         """Initialize an instance.
 
         Parameters
         ----------
-        *callbacks : `callable` or `Partial`'s
-            Partials to be called in sequence as listed.
+        *callbacks : `callable` or `SolverCallback`'s
```

Comment:
This is written as `callback1,...,callbackN` (not a hard rule, but the Numpy way; I think it makes more clear that the arguments are given individually, not packed)